### PR TITLE
[admin] Logs BSQL hangs

### PR DESCRIPTION
**Pull request by monster860**
Send a message to admins when BSQL hangs for more than half a second - close the question of if those freezes are being caused by BSQL or not.

### CODE REVIEW

File: code/modules/bsql/core/library.dm [hunk 1]
```diff
@@ -1,8 +1,12 @@
 /world/proc/_BSQL_Internal_Call(func, ...)
 	var/list/call_args = args.Copy(2)
+	var/start_time = world.tick_usage
 	BSQL_Debug("_BSQL_Internal_Call: [args[1]]([call_args.Join(", ")])")
 	. = call(_BSQL_Library_Path(), func)(arglist(call_args))
 	BSQL_Debug("Result: [. == null ? "NULL" : "\"[.]\""]")
+	var/ms = (world.tick_usage - start_time) * world.tick_lag
+	if(ms > 500)
+		message_admins("_BSQL_Internal_Call just hung for [ms]ms")
```
Comment (alexkar598) [inline code/modules/bsql/core/library.dm]:
```suggestion
		message_admins("_BSQL_Internal_Call just hung for [ms]ms: [args[1]]([call_args.Join(", ")])")
```

**Comment by Hopekz** [comment]
Should admins logs really be used to let admins know the server just lagged?
There is already quite a bit of information being spammed out about important events in-round and this just kind of adds spam to it

**Comment by Bibby0110** [comment]
> Should admins logs really be used to let admins know the server just lagged?
> There is already quite a bit of information being spammed out about important events in-round and this just kind of adds spam to it

this is just for 1 round while we fix stuff

**Comment by Hopekz** [comment]
> > Should admins logs really be used to let admins know the server just lagged?
> > There is already quite a bit of information being spammed out about important events in-round and this just kind of adds spam to it
> 
> this is just for 1 round while we fix stuff

Fair enough didn't get the memo.

File: code/modules/bsql/core/library.dm [hunk 2]
```diff
@@ -1,8 +1,12 @@
 /world/proc/_BSQL_Internal_Call(func, ...)
 	var/list/call_args = args.Copy(2)
+	var/start_time = world.tick_usage
 	BSQL_Debug("_BSQL_Internal_Call: [args[1]]([call_args.Join(", ")])")
 	. = call(_BSQL_Library_Path(), func)(arglist(call_args))
 	BSQL_Debug("Result: [. == null ? "NULL" : "\"[.]\""]")
+	var/ms = (world.tick_usage - start_time) * world.tick_lag
+	if(ms > 500)
+		message_admins("_BSQL_Internal_Call of [args[1]]([call_args.Join(", ")]) just hung for [ms]ms")
```
Comment (alexkar598) [inline code/modules/bsql/core/library.dm]:
```suggestion
		strack_trace("_BSQL_Internal_Call of [args[1]]([call_args.Join(", ")]) just hung for [ms]ms")
```

Comment (Bibby0110) [reply code/modules/bsql/core/library.dm]: stack_trace

File: code/modules/bsql/core/library.dm [hunk 3]
```diff
@@ -1,8 +1,12 @@
 /world/proc/_BSQL_Internal_Call(func, ...)
 	var/list/call_args = args.Copy(2)
+	var/start_time = world.tick_usage
 	BSQL_Debug("_BSQL_Internal_Call: [args[1]]([call_args.Join(", ")])")
 	. = call(_BSQL_Library_Path(), func)(arglist(call_args))
 	BSQL_Debug("Result: [. == null ? "NULL" : "\"[.]\""]")
+	var/ms = (world.tick_usage - start_time) * world.tick_lag
+	if(ms > 500)
+		strack_trace("_BSQL_Internal_Call of [args[1]]([call_args.Join(", ")]) just hung for [ms]ms")
```
Comment (alexkar598) [inline code/modules/bsql/core/library.dm]:
```suggestion
		stack_trace("_BSQL_Internal_Call of [args[1]]([call_args.Join(", ")]) just hung for [ms]ms")
```

**Pull request closed without merging**